### ISSUE TITLE
Reeder 3/iOS does not work with the fever API implementation

### ISSUE THREAD

**Issue by chemical1979** [FreshRSS/FreshRSS#1999]
I just installed FreshRSS master on my system to switch away from Feedly Cloud Solution to a self-hosted one. I'm pleasantly surprised how well the setup is described. Everything works out well.

I started with Unread on my iPad as a mobile reader which worked great, but isn't easy for me to adopt. So I started using [Reeder 3 (iOS)](https://itunes.apple.com/us/app/reeder-3/id697846300?mt=8) which is a better fit for me, but somehow does not work (Login failed). 

As Unread worked, I'm pretty sure my setup is correct. I also followed [this page](https://freshrss.github.io/FreshRSS/en/users/06_Fever_API.html) to check my setup. At the last part it hints about a change in fever.api (I guess it should read "/p/api/fever.php" instead) to log the requests that go in via API.

The only things that go into the log if I try to login with Reeder is this:

fever.log
`Reeder/3.1.21 CFNetwork/902.2 Darwin/17.7.0: {"api":""}`

There is not even content in the JSON requests, so I'm not sure whether the problem is within Reeder or within FreshRSS Fever API implementation. But as Reeder is free, maybe someone else more capable is able to help me here.

If I remove my login from Unread and re-add the source, this gets dumped into the log (comparision):

`Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":""}
Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":"","api_key":"409283a4322b2a61xxxxxxxxxxxxxxxx"}
Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":"","feeds":"","api_key":"409283a4322b2a61xxxxxxxxxxxxxxxx"}
Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":"","groups":"","api_key":"409283a4322b2a61xxxxxxxxxxxxxxxx"}
Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":"","unread_item_ids":"","api_key":"409283a4322b2a61xxxxxxxxxxxxxxxx"}
Unread/109055 CFNetwork/902.2 Darwin/17.7.0: {"api":"","saved_item_ids":"","api_key":"409283a4322b2a61xxxxxxxxxxxxxxxx"}`

Did someone positive-test Reeder 3 with FreshRSS? Maybe the problem is on my side.

**Comment by Alkarex** [comment]
Hello,
If you are on a 32-bit platform, it could be the same bug than https://github.com/FreshRSS/FreshRSS/issues/1962 , which is fixed in the /dev version of FreshRSS (upcoming version 1.11.2).
Please give it a try.

**Comment by chemical1979** [comment]
I noticed, while still on master branch, that Fiery Feeds (iOS app I tried later yesterday) "hangs" on a specific article and iterates always on the same id and loops every few seconds with the same request. From a first impression I got from #1962 , this seemed to be the case exactly.

I switched to dev branch (`git checkout -b dev origin/dev`) and the following things happened:

**Fiery Feeds** is now unable to login at all (worked with master branch). Ends after these API requests:
`Fiery/2.0.12 (iPad; iOS 11.4.1; Scale/2.00): {"api":"","groups":""}
Fiery/2.0.12 (iPad; iOS 11.4.1; Scale/2.00): {"api":"","feeds":""}
Fiery/2.0.12 (iPad; iOS 11.4.1; Scale/2.00): {"api":"","unread_item_ids":""}
Fiery/2.0.12 (iPad; iOS 11.4.1; Scale/2.00): {"api":"","saved_item_ids":""}
`
I tried to remove the account and re-enter it, but I was not able to get it working again. The situation persists. I re-set all passwords in via the web gui, but was not able to get Fiery to login again.

**Reeder** is still unable to login. Ends after these API requests:
`Reeder/3.1.21 CFNetwork/902.2 Darwin/17.7.0: {"api":""}
`

**Unread** still works as before (also after removing the account in the app and re-adding it).

Anyway, to sum it up: dev branch is still affected, Reeder is not able to login.

**Comment by chemical1979** [comment]
Interestingly, the login works when I change the hostname from `rss.mydomain.de/api/fever.php` to `https://rss.mydomain.de/api/fever.php` in Reeder "servername" field.

I need to dive into this further. Is FreshRSS aware whether a client came in via SSL or not?

**Comment by chemical1979** [comment]
It looks like Reeder defaults to http then and FreshRSS denies the api_key in the body of a non-https-request? At least this would explain it and I could see this behaviour in Postman also (auth:0 when POST came in via http, and it was 1 when it came via https).

However, as the issue is fixed with a changed connection string in the app, I close the issue.

**Comment by Alkarex** [comment]
Thank you for the additional debugging.
FreshRSS only detects HTTPS for cookie-related parameters, as well as for detecting its own URL at install time. This should not have any impact on the Fever API. Maybe there is a difference between your HTTP and HTTPS server configuration?

When using the correct address, is it working fine with the 3 clients (Fiery Fields, Reeder, Unread)?

**Comment by Alkarex** [comment]
Excellent, thanks for the feedback! Is Reeder working well enough to be added it to our list of compatible clients? (I do not have any Apple device to test)

* https://github.com/FreshRSS/FreshRSS/blob/dev/docs/en/users/06_Fever_API.md
* https://github.com/FreshRSS/FreshRSS/blob/dev/docs/fr/users/06_Fever_API.md
* https://github.com/FreshRSS/FreshRSS/blob/dev/README.md#fever-api
* https://github.com/FreshRSS/FreshRSS/blob/dev/README.fr.md#via-lapi-compatible-fever

**Comment by chemical1979** [comment]
I replaced Feedly App+Feedly Service with Reeder3+FreshRSS on personal RPi with SQLite and it works very well for me so far. So yes, you could list Reeder3 as a compatible app. 
I'm using dev branch currently, and you should either backport #1962 or release a new version really soon because that 32bit article id thing impacts nearly instantly, even if you have only a few feeds and articles. The apps "hang" in sync state and one with a less strong will to find the root cause will probably withdraw FreshRSS because of this.

Btw, slightly OT but is there any way to filter a feed (with intitle-similar features) or only display-wise in the webgui?

**Comment by Alkarex** [comment]
@chemical1979 Yes, FreshRSS 1.11.2 is scheduled for this week-end :-) https://github.com/FreshRSS/FreshRSS/pull/2001

**Comment by chemical1979** [comment]
> 1. With the user queries in FreshRSS, it is possible to save a search (either on a feed, a category, favourites, or everything) with positive as well as negative expressions (e.g. to exclude keywords)
> 2. I am planning to make a system to auto-mark-as-read some articles based on more powerful rules such as articles already existing in other feeds

To be honest, I mainly use FreshRSS as a backend via mobile apps. I assume the mentioned features are limited to the web GUI,  so the rules and filters do not have effects on any other client that uses the API.

Here is my use case for the filters: I have a stream with a lot of stuff, but only the articles with a specific word in the title are relevant for me.
In the web interface, there is the possibility of filters. But having the same feature request for every client could be circumvented by moving this filter a level lower so that the fetching of articles from the source already applies filters via e.g. SimplePie.

As this is really OT from this issue, I could create a new issue if you say this might be interesting.

**Comment by Alkarex** [comment]
Ok, and yes, please open another issue

**Comment by chemical1979** [comment]
@Alkarex As you're interested in compatible clients, FeedMe/Android does not seem to work anymore. I opened [an issue at their github Page](https://github.com/seazon/FeedMe/issues/67), but I'm not sure if this will work out.
